File: Java/557.java
__________________________________________________________________________________________________
sample 1 ms submission
class Solution {
    public String reverseWords(String s) {
        char[] c = s.toCharArray();
        int wordIndex = 0;
        int spaceIndex = 0;
        while ((spaceIndex = s.indexOf(" ", spaceIndex)) != -1) {
            reverse(c, wordIndex, spaceIndex-1);
            spaceIndex++;
            wordIndex = spaceIndex;
        }
        reverse(c, wordIndex, c.length - 1);
        return String.valueOf(c);
    }

    private static void reverse(char[] c, int i, int j) {
        while (i < j) {
            char temp = c[i];
            c[i] = c[j];
            c[j] = temp;
            i++;j--;
        }
    }
}
__________________________________________________________________________________________________
sample 37328 kb submission
class Solution {
    public String reverseWords(String s) {
        return String.join(" ",
                        Arrays.asList(s.split(" "))
                              .stream()
                              .map(sec->new StringBuilder(sec).reverse().toString())
                              .collect(Collectors.toList()));
    }
}
__________________________________________________________________________________________________

File: Python3/557.py
__________________________________________________________________________________________________
sample 20 ms submission
class Solution:
    def reverseWords(self, s: str) -> str:
        return " ".join(word[::-1] for word in s.split())
    
        sentence = ""
        for word in s.split():
            sentence += word[::-1] + " "
            
        return sentence.strip()
__________________________________________________________________________________________________
sample 13112 kb submission
class Solution:
    def reverseWords(self, s: str) -> str:
        a_list = s.split()
        res = ""
        for word in a_list:
            word = word[::-1]
            res += word + " "
        return res[:-1]
__________________________________________________________________________________________________
